ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -4180,6 +4180,7 @@ private RelNode genSelectForWindowing(QB qb, RelNode srcRel, HashSet<ColumnInfo>
 
       // 3. Construct new Row Resolver with everything from below.
       RowResolver out_rwsch = new RowResolver();
+      out_rwsch.setExprResolver(inputRR.getIsExprResolver());
```

Comment:
Should we set it to 'true' always instead of getting the value from inputRR...? If inputRR contains 'false' for 'isExprResolver' then we face same issue again right.

Reply:
IIUC there are two types of RowResolver: one can store expressions and the other one can't. This affects the ambiguity check. For the one which does not allow expression but columns only we need the ambiguity check. For the one which allows expressions like group by key duplicates are allowed even though it hasn't got an impact on the semantics.

Reply:
ok got your point, in the current flow `inputRR` related to group by for which `isExprResolver` is always true as per code flow so I feel the code changes are fine.